modules/mixins/cacheMixin.js

```diff
@@ -138,6 +140,7 @@
             options = options || {};
             //find ids that we don't have in cache
             options.idsToFetch = _.difference(this.getTrackedIds(), _.pluck(parentInstance.models, 'id'));
+            options.idsToFetch = _.difference(options.idsToFetch, _.uniq(_.flatten(_.values(parentInstance.idsFetching))));
```

Comment:
Isn't this line just overwriting what's in idsToFetch, so you should delete line 142?